File: shared/src/main/java/ru/privatenull/pnauth/command/CommandContext.java
package ru.privatenull.pnauth.command;

import java.util.List;

public record CommandContext(CommandSource source, String command, List<String> arguments) {
    public CommandContext {
        if (source == null) throw new IllegalArgumentException("source must not be null");
        command = command == null ? "" : command;
        arguments = arguments == null ? List.of() : List.copyOf(arguments);
    }
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/command/CommandContext.kt
package ru.privatenull.pnauth.command

@JvmRecord
data class CommandContext @JvmOverloads constructor(
    val source: CommandSource,
    val command: String = "",
    val arguments: List<String> = emptyList()
) {
    init {
        requireNotNull(source) { "source must not be null" }
    }
}
